pkg/planner/core/task.go

```diff
@@ -875,13 +911,100 @@ func (p *PhysicalTopN) getPushedDownTopN(childPlan base.PhysicalPlan) *PhysicalT
 	// Strictly speaking, for the row count of pushed down TopN, we should multiply newCount with "regionNum",
 	// but "regionNum" is unknown since the copTask can be a double read, so we ignore it now.
 	stats := util.DeriveLimitStats(childProfile, float64(newCount))
+
+	// Add a extra physicalProjection to save the distance column, a example like :
+	// select id from t order by vec_distance(vec, '[1,2,3]') limit x
+	// The Plan will be modified like:
+	//
+	// Original: DataSource(id, vec) -> TopN(by vec->dis) -> Projection(id)
+	//                                  └─Byitem: vec_distance(vec, '[1,2,3]')
+	//                                  └─Schema: id, vec
+	//
+	// New:      DataSource(id, vec) -> Projection(id, vec->dis) -> TopN(by dis) -> Projection(id)
```

Comment:
Seems that this comment is incorrect. Actually it is:

```
New: DataSource(id, vec) -> Projection(id, vec, vec->dis) -> TopN(by dis) -> Projection(id)
```

Reply:
This comment will fix in next cherry-pick~